Design.md

```diff
@@ -181,7 +181,9 @@ This API creates a user on Vault.
 1. The `get-policy` API will be called with the `userPolicy@[account-id]` policy name using assumed role credentials
 1. If the `get-policy` API does not return any policy, using the assumed role credentials creates an IAM managed policy with the `userPolicy@[account-id]` policy name with full S3 access.
 1. `attach-policy` for the new user will be called using the assumed role credentials and the policy arn formatted as `arn:aws:iam::[account-id]:policy/userPolicy@[account-id]`.
-1. `create-s3credentials` api will be called using assumed role credentials
+1. IAM `generate-access-key` api will be called using assumed role credentials
+1. Encrypt the `secret-key` using AES algorithm and store the encrypted `secret-key` on Redis Cache in the hash named as `OSIS` with the hash-key formatted as `<Username>__<AccessKeyID>`. 
+1. Log the `access-key-id` _(Log location and tools for logging are yet to be decided)_
```

Comment:
What are we referring to here as log location and tools?

Reply:
As per the @vrancurel 's suggestion above, I added this. We need to log the access key ids that we store on Vault using OSIS. This log file can be used to differentiate the keys created by OSIS and by other ways on Vault DB. This file is also useful when Redis crashes and we need to delete the keys created by OSIS.